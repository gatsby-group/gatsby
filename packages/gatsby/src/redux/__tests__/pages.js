const reducer = require(`../reducers/pages`)
const { actions } = require(`../actions`)

const start = Date.now()
Date.now = jest.fn(
  () =>
    // const diff = new Date().getTime() - start
    1482363367071 // + diff
)

describe(`Add pages`, () => {
  it(`allows you to add pages`, () => {
    const action = actions.createPage(
      {
        path: `/hi/`,
        component: `/whatever/index.js`,
      },
      { id: `test`, name: `test` }
    )
    const state = reducer(undefined, action)
    expect(action).toMatchSnapshot()
    expect(state).toMatchSnapshot()
  })

  it(`allows you to add pages with context`, () => {
    const action = actions.createPage(
      {
        path: `/hi/`,
        component: `/whatever/index.js`,
        context: {
          id: 123,
        },
      },
      { id: `test`, name: `test` }
    )
    const state = reducer(undefined, action)
    expect(action).toMatchSnapshot()
    expect(state).toMatchSnapshot()
  })

  it(`allows you to add multiple pages`, () => {
    const action = actions.createPage(
      {
        path: `/hi/`,
        component: `/whatever/index.js`,
      },
      { id: `test`, name: `test` }
    )
    const action2 = actions.createPage(
      {
        path: `/hi/pizza/`,
        component: `/whatever/index.js`,
      },
      { id: `test`, name: `test` }
    )
    let state = reducer(undefined, action)
    state = reducer(state, action2)
    expect(state).toMatchSnapshot()
    expect(state.length).toEqual(2)
  })

  it(`allows you to update existing pages (based on path)`, () => {
    const action = actions.createPage(
      {
        path: `/hi/`,
        component: `/whatever/index.js`,
      },
      { id: `test`, name: `test` }
    )

    // Change the component
    const action2 = actions.createPage(
      {
        path: `/hi/`,
        component: `/whatever2/index.js`,
      },
      { id: `test`, name: `test` }
    )

    let state = reducer(undefined, action)
    state = reducer(state, action2)
    expect(state).toMatchSnapshot()
    expect(state.length).toEqual(1)
  })

  it(`allows you to delete paths`, () => {
    const action = actions.createPage(
      {
        path: `/hi/`,
        component: `/whatever/index.js`,
      },
      { name: `test` }
    )
    const action2 = actions.deletePage({ path: `/hi/` })

    let state = reducer(undefined, action)
    state = reducer(state, action2)
    expect(state).toMatchSnapshot()
    expect(state.length).toEqual(0)
  })

  it(`allows you to add pages with multiple components`, () => {
    const action = actions.createPage(
      {
        path: `/hi/`,
        component: [`/whatever/index.js`, `/morewhatever/index.js`],
      },
      { id: `test`, name: `test` }
    )
    const state = reducer(undefined, action)
    expect(action).toMatchSnapshot()
    expect(state).toMatchSnapshot()
  })

  it(`allows you to add pages with context when using multiple components`, () => {
    const action = actions.createPage(
      {
        path: `/hi/`,
        component: [`/whatever/index.js`, `/differentwhatever/index.js`],
        context: {
          id: 123,
        },
      },
      { id: `test`, name: `test` }
    )
    const state = reducer(undefined, action)
    expect(action).toMatchSnapshot()
    expect(state).toMatchSnapshot()
  })

  it(`allows you to add multiple pages each with multiple components`, () => {
    const action = actions.createPage(
      {
        path: `/hi/`,
        component: [`/whatever/index.js`, `/morewhatever/index.js`],
      },
      { id: `test`, name: `test` }
    )
    const action2 = actions.createPage(
      {
        path: `/hi/hotpizza/`,
        component: [`/whatever/index.js`, `/pizza/index.js`],
      },
      { id: `test`, name: `test` }
    )
    let state = reducer(undefined, action)
    state = reducer(state, action2)
    expect(state).toMatchSnapshot()
    expect(state.length).toEqual(2)
  })

  it(`allows you to update existing pages with multiple components (based on path)`, () => {
    const action = actions.createPage(
      {
        path: `/hi/`,
        component: [`/something/index.js`, `/morewhatever/index.js`],
      },
      { id: `test`, name: `test` }
    )

    // Change the component
    const action2 = actions.createPage(
      {
        path: `/hi/`,
        component: [`/whateven/index.js`, `/supercereal/index.js`],
      },
      { id: `test`, name: `test` }
    )

    let state = reducer(undefined, action)
    state = reducer(state, action2)
    expect(state).toMatchSnapshot()
    expect(state.length).toEqual(1)
  })

  it(`allows you to delete paths that have multiple components`, () => {
    const action = actions.createPage(
      {
        path: `/hi/`,
        component: [`/ohhey/index.js`, `/goodbye/index.js`],
      },
      { name: `test` }
    )
    const action2 = actions.deletePage({ path: `/hi/` })

    let state = reducer(undefined, action)
    state = reducer(state, action2)
    expect(state).toMatchSnapshot()
    expect(state.length).toEqual(0)
  })
})
