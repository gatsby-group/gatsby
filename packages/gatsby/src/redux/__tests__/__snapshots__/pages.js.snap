// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add pages allows you to add multiple pages 1`] = `
Array [
  Object {
    "component": Array [
      "/whatever/index.js",
    ],
    "componentChunkName": Array [
      "component---whatever-index-js",
    ],
    "context": Object {},
    "internalComponentName": "ComponentHi",
    "jsonName": "hi.json",
    "path": "/hi/",
    "pluginCreatorId": "test",
    "pluginCreator___NODE": "test",
    "updatedAt": 1482363367071,
  },
  Object {
    "component": Array [
      "/whatever/index.js",
    ],
    "componentChunkName": Array [
      "component---whatever-index-js",
    ],
    "context": Object {},
    "internalComponentName": "ComponentHiPizza",
    "jsonName": "hi-pizza.json",
    "path": "/hi/pizza/",
    "pluginCreatorId": "test",
    "pluginCreator___NODE": "test",
    "updatedAt": 1482363367071,
  },
]
`;

exports[`Add pages allows you to add multiple pages each with multiple components 1`] = `
Array [
  Object {
    "component": Array [
      "/whatever/index.js",
      "/morewhatever/index.js",
    ],
    "componentChunkName": Array [
      "component---whatever-index-js",
      "component---morewhatever-index-js",
    ],
    "context": Object {},
    "internalComponentName": "ComponentHi",
    "jsonName": "hi.json",
    "path": "/hi/",
    "pluginCreatorId": "test",
    "pluginCreator___NODE": "test",
    "updatedAt": 1482363367071,
  },
  Object {
    "component": Array [
      "/whatever/index.js",
      "/pizza/index.js",
    ],
    "componentChunkName": Array [
      "component---whatever-index-js",
      "component---pizza-index-js",
    ],
    "context": Object {},
    "internalComponentName": "ComponentHiHotpizza",
    "jsonName": "hi-hotpizza.json",
    "path": "/hi/hotpizza/",
    "pluginCreatorId": "test",
    "pluginCreator___NODE": "test",
    "updatedAt": 1482363367071,
  },
]
`;

exports[`Add pages allows you to add pages 1`] = `
Object {
  "payload": Object {
    "component": Array [
      "/whatever/index.js",
    ],
    "componentChunkName": Array [
      "component---whatever-index-js",
    ],
    "context": Object {},
    "internalComponentName": "ComponentHi",
    "jsonName": "hi.json",
    "path": "/hi/",
    "pluginCreatorId": "test",
    "pluginCreator___NODE": "test",
    "updatedAt": 1482363367071,
  },
  "plugin": Object {
    "id": "test",
    "name": "test",
  },
  "traceId": undefined,
  "type": "CREATE_PAGE",
}
`;

exports[`Add pages allows you to add pages 2`] = `
Array [
  Object {
    "component": Array [
      "/whatever/index.js",
    ],
    "componentChunkName": Array [
      "component---whatever-index-js",
    ],
    "context": Object {},
    "internalComponentName": "ComponentHi",
    "jsonName": "hi.json",
    "path": "/hi/",
    "pluginCreatorId": "test",
    "pluginCreator___NODE": "test",
    "updatedAt": 1482363367071,
  },
]
`;

exports[`Add pages allows you to add pages with context 1`] = `
Object {
  "payload": Object {
    "component": Array [
      "/whatever/index.js",
    ],
    "componentChunkName": Array [
      "component---whatever-index-js",
    ],
    "context": Object {
      "id": 123,
    },
    "internalComponentName": "ComponentHi",
    "jsonName": "hi.json",
    "path": "/hi/",
    "pluginCreatorId": "test",
    "pluginCreator___NODE": "test",
    "updatedAt": 1482363367071,
  },
  "plugin": Object {
    "id": "test",
    "name": "test",
  },
  "traceId": undefined,
  "type": "CREATE_PAGE",
}
`;

exports[`Add pages allows you to add pages with context 2`] = `
Array [
  Object {
    "component": Array [
      "/whatever/index.js",
    ],
    "componentChunkName": Array [
      "component---whatever-index-js",
    ],
    "context": Object {
      "id": 123,
    },
    "internalComponentName": "ComponentHi",
    "jsonName": "hi.json",
    "path": "/hi/",
    "pluginCreatorId": "test",
    "pluginCreator___NODE": "test",
    "updatedAt": 1482363367071,
  },
]
`;

exports[`Add pages allows you to add pages with context when using multiple components 1`] = `
Object {
  "payload": Object {
    "component": Array [
      "/whatever/index.js",
      "/differentwhatever/index.js",
    ],
    "componentChunkName": Array [
      "component---whatever-index-js",
      "component---differentwhatever-index-js",
    ],
    "context": Object {
      "id": 123,
    },
    "internalComponentName": "ComponentHi",
    "jsonName": "hi.json",
    "path": "/hi/",
    "pluginCreatorId": "test",
    "pluginCreator___NODE": "test",
    "updatedAt": 1482363367071,
  },
  "plugin": Object {
    "id": "test",
    "name": "test",
  },
  "traceId": undefined,
  "type": "CREATE_PAGE",
}
`;

exports[`Add pages allows you to add pages with context when using multiple components 2`] = `
Array [
  Object {
    "component": Array [
      "/whatever/index.js",
      "/differentwhatever/index.js",
    ],
    "componentChunkName": Array [
      "component---whatever-index-js",
      "component---differentwhatever-index-js",
    ],
    "context": Object {
      "id": 123,
    },
    "internalComponentName": "ComponentHi",
    "jsonName": "hi.json",
    "path": "/hi/",
    "pluginCreatorId": "test",
    "pluginCreator___NODE": "test",
    "updatedAt": 1482363367071,
  },
]
`;

exports[`Add pages allows you to add pages with multiple components 1`] = `
Object {
  "payload": Object {
    "component": Array [
      "/whatever/index.js",
      "/morewhatever/index.js",
    ],
    "componentChunkName": Array [
      "component---whatever-index-js",
      "component---morewhatever-index-js",
    ],
    "context": Object {},
    "internalComponentName": "ComponentHi",
    "jsonName": "hi.json",
    "path": "/hi/",
    "pluginCreatorId": "test",
    "pluginCreator___NODE": "test",
    "updatedAt": 1482363367071,
  },
  "plugin": Object {
    "id": "test",
    "name": "test",
  },
  "traceId": undefined,
  "type": "CREATE_PAGE",
}
`;

exports[`Add pages allows you to add pages with multiple components 2`] = `
Array [
  Object {
    "component": Array [
      "/whatever/index.js",
      "/morewhatever/index.js",
    ],
    "componentChunkName": Array [
      "component---whatever-index-js",
      "component---morewhatever-index-js",
    ],
    "context": Object {},
    "internalComponentName": "ComponentHi",
    "jsonName": "hi.json",
    "path": "/hi/",
    "pluginCreatorId": "test",
    "pluginCreator___NODE": "test",
    "updatedAt": 1482363367071,
  },
]
`;

exports[`Add pages allows you to delete paths 1`] = `Array []`;

exports[`Add pages allows you to delete paths that have multiple components 1`] = `Array []`;

exports[`Add pages allows you to update existing pages (based on path) 1`] = `
Array [
  Object {
    "component": Array [
      "/whatever2/index.js",
    ],
    "componentChunkName": Array [
      "component---whatever-2-index-js",
    ],
    "context": Object {},
    "internalComponentName": "ComponentHi",
    "jsonName": "hi.json",
    "path": "/hi/",
    "pluginCreatorId": "test",
    "pluginCreator___NODE": "test",
    "updatedAt": 1482363367071,
  },
]
`;

exports[`Add pages allows you to update existing pages with multiple components (based on path) 1`] = `
Array [
  Object {
    "component": Array [
      "/whateven/index.js",
      "/supercereal/index.js",
    ],
    "componentChunkName": Array [
      "component---whateven-index-js",
      "component---supercereal-index-js",
    ],
    "context": Object {},
    "internalComponentName": "ComponentHi",
    "jsonName": "hi.json",
    "path": "/hi/",
    "pluginCreatorId": "test",
    "pluginCreator___NODE": "test",
    "updatedAt": 1482363367071,
  },
]
`;
